lib/register.ts

```diff
@@ -140,6 +140,8 @@ export const getRegisterDevice = ({
 				os_variant: options.osVariant,
 				mac_address: options.macAddress,
 			},
+		}).catch(e => {
+			throw new Error(`Device register request.send failed ${e}`);
 		});
```

Comment:
This should use `try/catch` instead of `.catch`, but aside from that what was the issue you saw?

Reply:
@Page- Not throwing this error was masquerading an error message on failing DNS resolution with an empty error message. I found this while debugging balenaOS in docker  to connect to BoB which failed b/c of DNS resolution, but just through an empty '' error message to the supervisor logs.

Reply:
I've just realised that the tests fail and came to the conclusion that a proper error should be through in the passed request.send method. This PR is obsolete.